mathesar/models/query.py

```diff
@@ -343,8 +340,31 @@ def _get_display_name_for_alias(self, alias):
         return self._alias_to_display_name.get(alias)
 
     def _get_display_options_for_alias(self, alias):
-        if self.display_options is not None:
-            return self.display_options.get(alias)
+        display_options = None
+        if self.display_options:
+            display_options = self.display_options.get(alias)
+        if display_options is None:
+            # notice that this isn't meant to support non-initial-column aliases
+            dj_col = self._map_of_initial_col_alias_to_dj_column.get(alias)
+            if dj_col:
+                display_options = dj_col.display_options
+        return display_options
+
+    @property
+    def _map_of_initial_col_alias_to_dj_column(self):
+        dj_column_ids = [col['id'] for col in self.initial_columns]
+        dj_columns = Column.objects.filter(pk__in=dj_column_ids)
+        initial_col_aliases = [
+            initial_col['alias']
+            for initial_col
+            in self.initial_columns
+        ]
+        return frozendict(
```

Comment:
What is the reason to use a `frozendict` here?

Reply:
Just immutability. Could be converted into `@cached_property` without changes to the method body, for example.